### PR TITLE
Add support for full tflite deployments

### PR DESCRIPTION
**Pull request by faradayberry**
Added support for full tflite deployments as tested with object detection and image classification models, as well as checking for YOLOV5 to add appropriate suffix in ei_fill_result_struct.h and fix [issue #11167](https://github.com/edgeimpulse/edgeimpulse/issues/11167). This includes updates to:

EIDownload.py
- Added print of error response for easier troubleshooting

generate.py
- Added `--full-tflite` argument to use tflite instead of default EON
- Updated **edit_file** function to handle includes
- Added a function **check_for_yolov5** to return `True` if any model data indicates YOLOV5 and if so an additional modification is made in ei_fill_result_struct.h to append the project suffix to pattern _ei_classifier_inferencing_categories_
- Added a function **insert_define_statement** to insert a new #define after the #includes and before the rest of any #defines
- Added functions **insert_after_line**, **replace_line**, and **remove_line** to perform these operations in some files as required by the full tflite changes
- Added a function **merge_tflite_full** to merge the projects' tflite_full.h #includes for pattern _trained_model_ops_define_
- Added an `else` to `if "compiled" in f:` to cover case when built files don't have "compiled" in the name

In main loop added check for tflite and made additional updates:
- Insert #define EI_CLASSIFIER_USE_FULL_TFLITE 1 in model_metadata.h
- Remove deprecated API reference in tflite_full.h
- Edit Zephyr cmake to use full tflite

At the end added use of **merge_tflite_full** to merge all the #includes for _trained_model_ops_define_

### CODE REVIEW

**Comment by luisomoreau** [comment]
I saw plenty of issues, including the object detection support when it is not the first impulse to be downloaded and merged.
I created this other PR from this one that uses a custom_display function to avoid the `#if` statement that are not available from the merged `model_metadata.h`. e.g `#define EI_CLASSIFIER_OBJECT_DETECTION             0`

https://github.com/edgeimpulse/multi-impulse-deployment-block/pull/5

**Comment by AlessandroEdge** [comment]
Is this ready to be merged and will it address https://github.com/edgeimpulse/edgeimpulse/issues/11167?

**Comment by faradayberry** [comment]
> Is this ready to be merged and will it address [edgeimpulse/edgeimpulse#11167](https://github.com/edgeimpulse/edgeimpulse/issues/11167)?

I haven't made any changes to address @luisomoreau concerns from above, I can only confirm it works for the combination I needed it for, an object detector as the main project (with sometimes deploying YOLOV5, other times TAO) and an image classifier as the other project.

**Comment by AlessandroEdge** [comment]
Ok let's wait for Louis to be back

**Comment by AIWintermuteAI** [comment]
Okay, should have pushed my changes earlier...
I started working on multi-impulse block fixes last week, as per internal repo issue #11564.
Few comments:
> Added --full-tflite argument to use tflite instead of default EON

Should just be --tflite or --non-eon. 
> Added a function check_for_yolov5 to return True if any model data indicates YOLOV5 and if so an additional modification is made in ei_fill_result_struct.h to append the project suffix to pattern ei_classifier_inferencing_categories

This needs to be fixed in SDK. When it is fixed we can remove YOLOv5 check.

I'll take a look at other changes.

File: generate.py [hunk 1]
```diff
@@ -224,23 +415,56 @@ def merge_model_ops(src_file, dest_file):
     ]
     edit_file(f, patterns, suffix)
 
-    # 3. Merge model_variables.h into 1st project
+    # 3. Check if full tflite and make additional required modifications
+    if args.full_tflite:
+        # Insert #define EI_CLASSIFIER_USE_FULL_TFLITE 1 in model_metadata.h
+        f = os.path.join(tmpdir, p, "model-parameters/model_metadata.h")
+        insert_define_statement(f, "#define EI_CLASSIFIER_USE_FULL_TFLITE 1")
```
Comment (AIWintermuteAI) [inline generate.py:422]: This should not be inserted into model_metadata.h, but instead passed in Makefile or CMakeLists, as done here
https://github.com/edgeimpulse/example-standalone-inferencing-linux/blob/95675547fae720ab2adf7e514f89dfe231a1c0b1/Makefile#L81

File: generate.py [hunk 2]
```diff
@@ -224,23 +415,56 @@ def merge_model_ops(src_file, dest_file):
...
+
+        # Remove deprecated API reference in tflite_full.h
```
Comment (AIWintermuteAI) [inline generate.py:424]: It is not deprecated, afaik.

File: generate.py [hunk 3]
```diff
@@ -224,23 +415,56 @@ def merge_model_ops(src_file, dest_file):
...
+        f = os.path.join(tmpdir, p, "edge-impulse-sdk/classifier/inferencing_engines/tflite_full.h")
+        str_rm = 'tree_ensemble_classifier'
+        remove_line(f, str_rm)
+
+        # Edit Zephyr cmake to use full tflite
```
Comment (AIWintermuteAI) [inline generate.py:429]: I don't understand why this is needed. Zephyr uses full tflite???

File: generate.py [hunk 4]
```diff
@@ -174,16 +185,181 @@ def merge_model_ops(src_file, dest_file):
...
+            file.writelines(file_content)
+
+        print(f"Removed line containing {search_string} from {file_path}")
+
+    except FileNotFoundError:
+        print(f"File not found: {file_path}")
+    except Exception as e:
+        print(f"An error occurred: {str(e)}")
+
+# Merge the tflite_full.h header files to include all projects
```
Comment (AIWintermuteAI) [inline generate.py:307]: Again, one SHALL NOT patch the EI SDK here unless absolutely necessary. Why not simply merge the two trained_model_ops_define?

**Pull request merged**